src/prover_disk.hpp

```diff
@@ -635,11 +639,18 @@ class DiskProver {
     // all of the leaves (x values). For example, for depth=5, it fetches the position-th
     // entry in table 5, reading the two back pointers from the line point, and then
     // recursively calling GetInputs for table 4.
-    std::vector<Bits> GetInputs(uint64_t position, uint8_t depth)
+    std::vector<Bits> GetInputs(uint64_t position, uint8_t depth, std::ifstream* disk_file=nullptr)
     {
-        // Create individual file handles to allow parallel processing
-        std::ifstream disk_file(filename, std::ios::in | std::ios::binary);
-        uint128_t line_point = ReadLinePoint(disk_file, depth, position);
+        uint128_t line_point;
+
+        if (!disk_file) {
```

Comment:
```suggestion
        if (disk_file == nullptr) {
```
Clearer to read imo.

Reply:
I'm not opposed to this - initially I had it the other way . Personally, I find it slightly more obvious this way that you are disabling the feature. I assume you would want this then also in the config.yaml as "parallel_reads: false".

Reply:
This particular diff doesn't change any behavior, its just "be explicit in the condition" i.e. `!x` has the same outcome as `x == nullptr`

> I'm not opposed to this - initially I had it the other way . Personally, I find it slightly more obvious this way that you are disabling the feature.

I guess you are talking about the renaming of the flag here? If so:

> I assume you would want this then also in the config.yaml as "parallel_reads: false".

Yeah :+1: that's what i meant with: 

> Same for the `chia-blockchain` PR.